File: templates/conteudo/capa-reel/render.js
// render.js — renderiza capa-reel.html em PNG 1080x1920
// Uso: node render.js <html> <output.png>
// Ex:  node render.js capa-reel.html marketing/conteudo/reel-xxx/capa.png

const { chromium } = require('playwright');
const path = require('path');

async function render(htmlFile, outputFile) {
  const browser = await chromium.launch();
  const page = await browser.newPage();
  await page.setViewportSize({ width: 1080, height: 1920 });
  await page.goto(`file://${path.resolve(htmlFile)}`);
  await page.waitForTimeout(800); // aguarda Poppins carregar
  const slide = await page.$('.slide');
  await slide.screenshot({ path: outputFile });
  await browser.close();
  console.log(`✓ ${outputFile}`);
}

const htmlFile = process.argv[2] || 'capa-reel.html';
const outputFile = process.argv[3] || 'capa-reel.png';
render(htmlFile, outputFile).catch(err => { console.error(err.message); process.exit(1); });

File: marketing/conteudo/reel-auriculoterapia-2026-07-22/render.py
import asyncio
from playwright.async_api import async_playwright

URL = "http://localhost:7823/marketing/conteudo/reel-auriculoterapia-2026-07-22/capa.html"
OUT = "marketing/conteudo/reel-auriculoterapia-2026-07-22/capa.png"

async def main():
    async with async_playwright() as p:
        browser = await p.chromium.launch()
        page = await browser.new_page(viewport={"width": 1080, "height": 1920}, device_scale_factor=1)
        await page.goto(URL, wait_until="networkidle")
        await page.wait_for_timeout(1200)
        await page.locator(".slide").screenshot(path=OUT)
        await browser.close()
        print("done ->", OUT)

asyncio.run(main())
